movement/kinematics.py

```diff
@@ -324,6 +326,330 @@ def compute_head_direction_vector(
+
+    Compute the Euclidean distance (default) between ``key1`` and
+    ``key2`` (i.e. interkeypoint distance for all individuals)
+    using the ``position`` data variable in the Dataset ``ds``:
+
+    >>> pos1 = ds.position.sel(keypoints="key1")
+    >>> pos2 = ds.position.sel(keypoints="key2")
+    >>> key_dists = _cdist(pos1, pos2, dim="keypoints")
+
+    See Also
+    --------
+    scipy.spatial.distance.cdist : The underlying function used.
+    compute_pairwise_distances : Compute pairwise distances between
+        ``individuals`` or ``keypoints``
+
+    """
+    # The dimension from which ``dim`` labels are obtained
+    labels_dim = "individuals" if dim == "keypoints" else "keypoints"
+    elem1 = getattr(a, dim).item()
+    elem2 = getattr(b, dim).item()
+    a = _validate_labels_dimension(a, labels_dim)
+    b = _validate_labels_dimension(b, labels_dim)
+    result = xr.apply_ufunc(
+        cdist,
+        a,
+        b,
+        kwargs={"metric": metric, **kwargs},
+        input_core_dims=[[labels_dim, "space"], [labels_dim, "space"]],
+        output_core_dims=[[elem1, elem2]],
+        vectorize=True,
+    )
+    result = result.assign_coords(
+        {
+            elem1: getattr(a, labels_dim).values,
+            elem2: getattr(a, labels_dim).values,
+        }
+    )
+    # Drop any squeezed coordinates
+    return result.squeeze(drop=True)
+
+
+def compute_pairwise_distances(
+    data: xr.DataArray,
+    dim: Literal["individuals", "keypoints"],
+    pairs: dict[str, str | list[str]] | Literal["all"],
+    metric: str | None = "euclidean",
+    **kwargs,
+) -> xr.DataArray | dict[str, xr.DataArray]:
+    """Compute pairwise distances between ``individuals`` or ``keypoints``.
+
+    This function computes the distances between
+    pairs of ``individuals`` (i.e. interindividual distances) or
+    pairs of ``keypoints`` (i.e. interkeypoint distances),
+    as determined by ``dim``.
+    The distances are computed for the given ``pairs``
+    using the specified ``metric``.
+
+    Parameters
+    ----------
+    data : xarray.DataArray
+        The input data containing
+        ``('time', 'individuals', 'keypoints', 'space')`` as dimensions.
+    dim : Literal["individuals", "keypoints"]
+        The dimension to compute the distances for. Must be either
+        ``'individuals'`` or ``'keypoints'``.
+    pairs : dict[str, str | list[str]] | Literal["all"]
+        A dictionary containing the mapping between pairs of ``individuals``
+        or ``keypoints``, or the special keyword ``"all"``.
+
+        - If a dictionary is provided,
+          the key is the keypoint or individual and
+          the value can be a list of keypoints or individuals,
+          or a string representing a single keypoint or individual
+          to compute the distance with.
+        - If the special keyword ``'all'`` is provided,
+          all possible combinations of pairs are computed.
```

Comment:
I suggest a more verbose description for this parameter, to clarify the exact interdependence between `pairs` and `dim`:

```suggestion
    pairs : dict[str, str | list[str]] or 'all'
        Specifies the pairs of elements (either individuals or keypoints)
        for which to compute distances, depending on the value of ``dim``.

        - If ``dim='individuals'``, ``pairs`` should be a dictionary where
          each key is an individual name, and each value is also an individual
          name or a list of such names to compute distances with.
        - If ``dim='keypoints'``, ``pairs`` should be a dictionary where each
          key is a keypoint name, and each value is also keypoint name or a
          list of such names to compute distances with.
        - Alternatively, use the special keyword ``'all'`` to compute distances
          for all possible pairs of individuals or keypoints
          (depending on ``dim``).
```